node/builder_miner.go

```diff
@@ -62,29 +50,14 @@ func ConfigStorageMiner(c interface{}) Option {
 		return Error(xerrors.Errorf("invalid config from repo, got: %T", c))
 	}
 
-	pricingConfig := cfg.Dealmaking.RetrievalPricing
-	if pricingConfig.Strategy == config.RetrievalPricingExternalMode {
-		if pricingConfig.External == nil {
-			return Error(xerrors.New("retrieval pricing policy has been to set to external but external policy config is nil"))
-		}
-
-		if pricingConfig.External.Path == "" {
-			return Error(xerrors.New("retrieval pricing policy has been to set to external but external script path is empty"))
-		}
-	} else if pricingConfig.Strategy != config.RetrievalPricingDefaultMode {
-		return Error(xerrors.New("retrieval pricing policy must be either default or external"))
-	}
-
-	enableLibp2pNode := cfg.Subsystems.EnableMarkets // we enable libp2p nodes if the storage market subsystem is enabled, otherwise we don't
-
 	return Options(
 
 		Override(new(v1api.FullNode), modules.MakeUuidWrapper),
 		// Needed to instantiate pubsub used by index provider via ConfigCommon
 		Override(new(dtypes.DrandSchedule), modules.BuiltinDrandConfig),
 		Override(new(dtypes.BootstrapPeers), modules.BuiltinBootstrap),
 		Override(new(dtypes.DrandBootstrap), modules.DrandBootstrap),
-		ConfigCommon(&cfg.Common, enableLibp2pNode),
+		ConfigCommon(&cfg.Common, cfg.EnableLibp2p),
```

Comment:
```suggestion
		ConfigCommon(&cfg.Common, false),
```

As per comment below, I don't think we need this option, it doesn't need a host to listen for anything anymore does it?

Reply:
@rvagg We need this flag as enabling this flag allows Miners to talk to the deamon over libp2p in the testkit.

Reply:
don't they use RPC though? the miners need to talk together but I thought the rest was all just RPC